sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -408,9 +468,42 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
     mapboxRenderer.map = null
   }
 
+  private fun clearQueue(lock: ReentrantLock, queue: LinkedList<RenderEvent>) {
+    // we clean only Mapbox events to avoid outdated runnables associated with previous EGL context
+    lock.withLock {
+      // using iterator to avoid concurrent modification exception
```

Comment:
Could just use kotlin ext `MutableList<T>.removeAll(predicate: (T) -> Boolean)`

Reply:
as said before in other place - requires API = 24

Reply:
Not really, API 24 is required for java member function 
https://developer.android.com/reference/java/util/Collection#removeIf(java.util.function.Predicate%3C?%20super%20E%3E)
but there's Kotlin extension available for any API lvl
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/remove-all.html